### PR TITLE
inherit compileSdkVersion from project

### PR DESCRIPTION
**Pull request by anthonyhumphreys**
Relate to: [this issue](https://github.com/aws-amplify/amplify-js/issues/4360) - few more examples from googling.

I've updated the build.gradle for the push notification module to inherit the compileSdkVersion from the root project - not sure if this needs some safety around it like a fallback version.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by amhinson** [comment]
Hey @anthonyhumphreys Thanks for the PR!

I'm thinking that it might be a bit safer to have some defaults as well, just in case. Maybe we could do something similar to [this example](https://github.com/react-native-community/react-native-netinfo/blob/master/android/build.gradle#L28) where there are some helper methods that check for the root project versions but fallback to versions defined in a `gradle.properties`?

**Comment by anthonyhumphreys** [comment]
Great idea – I’ll update my PR.

I knew I’d seen that somewhere but couldn’t find the repo!

Anthony Humphreys | Full Stack ITPI Developer
ISS | Lancaster University

Pronouns: He/Him/His
Working hours: Mondays, Tuesdays and Wednesdays 9am – 5:30pm; Thursday and Friday 9am-5pm

www.lancaster.ac.uk<https://www.lancaster.ac.uk/itpi>

[Lancaster University]<https://www.lancaster.ac.uk/>

[Facebook]<https://www.facebook.com/lancasteruniversity/> [Instagram] <https://www.instagram.com/lancasteriss>  [Twitter] <https://www.twitter.com/ISSLancaster>  [LinkedIn] <https://www.linkedin.com/in/anthonyphumphreys/>

**Comment by anthonyhumphreys** [comment]
Hey, I've updated my PR - not sure what a sensible fallback value is... is there a wildcard for 'latest'?

**Comment by amhinson** [comment]
I think we can set both the default `compileSdkVersion` & `targetSdkVersion` at 28 if the root project doesn't have it defined. Does that sound good to you?

**Comment by anthonyhumphreys** [comment]
Sounds like a plan - also, looks like I forgot to push 🤦‍♂️

**Comment by anthonyhumphreys** [comment]
There we go 😆 this should work 

https://github.com/aws-amplify/amplify-js/pull/5475/commits/f204bd02b947f8e441fcd71099e486aaeb820d6c

File: packages/pushnotification/android/build.gradle [hunk 1]
```diff
@@ -19,15 +19,19 @@ allprojects {
     }
 }
 
+def getExtOrIntegerDefault(name) {
+  return rootProject.ext.has(name) ? rootProject.ext.get(name) : 28
+}
+
```
Comment (amhinson) [inline packages/pushnotification/android/build.gradle:25]: Can we make this function a bit more generic to handle setting all of the version values?
```suggestion
def getExtOrDefault(name, default) {
  return rootProject.ext.has(name) ? rootProject.ext.get(name) : default
}
```

and then we can use it below for all of the versions:
```gradle
compileSdkVersion getExtOrDefault('compileSdkVersion', 28) 

buildToolsVersion getExtOrDefault('buildToolsVersion', "28.0.3") 

minSdkVersion getExtOrDefault('minSdkVersion', 16) 

targetSdkVersion getExtOrDefault('targetSdkVersion', 28) 
```

Also (as noted in the code above), we should bump up the `buildToolsVersion` as well.

Comment (anthonyhumphreys) [reply packages/pushnotification/android/build.gradle:25]: Yeah great suggestion, that slipped me by. Will fix.

**Comment by anthonyhumphreys** [comment]
Should be good to go now @amhinson :) - I hope! Thanks for your patience

**Comment by anthonyhumphreys** [comment]
Cool, i've moved those inside the defaultConfig block now :)

File: packages/pushnotification/android/build.gradle [hunk 2]
```diff
@@ -19,15 +19,18 @@ allprojects {
...
-    compileSdkVersion 26
-    buildToolsVersion "26.0.3"
-
     defaultConfig {
-        minSdkVersion 16
-        targetSdkVersion 26
+        compileSdkVersion getExtOrIntegerDefault('compileSdkVersion', 28)
+        buildToolsVersion getExtOrDefault('buildToolsVersion', "28.0.3")
+        minSdkVersion getExtOrDefault('minSdkVersion', 16)
+        targetSdkVersion getExtOrDefault('targetSdkVersion', 28)
```
Comment (amhinson) [inline packages/pushnotification/android/build.gradle:33]: Sorry, I think we need one more change and then it should be good to go. Only `minSdkVersion` and `targetSdkVersion` should be inside `defaultConfig`. The other 2 can stay at the root level inside `android`.

Comment (anthonyhumphreys) [reply packages/pushnotification/android/build.gradle]: Gotcha :)

**Pull request merged**